src/hooks/users.ts

```diff
@@ -1,63 +1,159 @@
+
+    const relatedLocationIds = [currentVenue.id];
+
+    if (currentVenue.parentId) {
+      relatedLocationIds.push(currentVenue.parentId);
+    }
+
+    return [
+      {
+        collection: "users",
+        where: ["enteredVenueIds", "array-contains-any", relatedLocationIds],
+        storeAs: "worldUsers",
+      },
+    ];
+  });
+};
+
+export const useWorldUsers = (): {
+  worldUsers: readonly WithId<User>[];
+  isWorldUsersLoaded: boolean;
+} => {
+  useConnectWorldUsers();
+
+  const selectedWorldUsers = useSelector(worldUsersSelector);
+
+  return useMemo(
+    () => ({
+      worldUsers: selectedWorldUsers ?? [],
+      isWorldUsersLoaded: isLoaded(selectedWorldUsers),
+    }),
+    [selectedWorldUsers]
   );
 };
 
-export const useVenueUsers = (): readonly WithId<User>[] => {
-  useConnectVenueUsers();
+export const useWorldUsersById = () => {
+  useConnectWorldUsers();
+
+  const worldUsersById = useSelector(worldUsersByIdSelector);
 
-  return useSelector(usersSelector) ?? [];
+  return useMemo(
+    () => ({
+      worldUsersById: worldUsersById ?? {},
+      isWorldUsersLoaded: isLoaded(worldUsersById),
+    }),
+    [worldUsersById]
+  );
 };
 
-export const usePartygoers = (): readonly WithId<User>[] => {
+export const useRecentWorldsUsers = (): {
+  recentWorldUsers: readonly WithId<User>[];
+  isRecentWorldUsersLoaded: boolean;
+} => {
   const lastSeenThreshold = useUserLastSeenThreshold();
 
-  const venueUsers = useVenueUsers();
+  const { worldUsers, isWorldUsersLoaded } = useWorldUsers();
 
   return useMemo(
-    () => venueUsers.filter((user) => user.lastSeenAt > lastSeenThreshold),
-    [venueUsers, lastSeenThreshold]
+    () => ({
+      recentWorldUsers: worldUsers.filter(
+        (user) => user.lastSeenAt > lastSeenThreshold
+      ),
+      isRecentWorldUsersLoaded: isWorldUsersLoaded,
+    }),
+    [worldUsers, isWorldUsersLoaded, lastSeenThreshold]
   );
 };
 
-export const useIsVenueUsersLoaded = () => {
-  // @debt The reason for this useConnect to be here is that we have entry point components,
-  // which have useConnects calls. And there are checks for the data loaded statuses.
-  // We want to gradualy move from that approach to a more modular one, where the specific data is connected where it is required
-  useConnectVenueUsers();
+export const useRecentVenueUsers = (): {
+  recentVenueUsers: readonly WithId<User>[];
+  isRecentVenueUsersLoaded: boolean;
+} => {
+  const lastSeenThreshold = useUserLastSeenThreshold();
+  const venueId = useVenueId();
+
+  const { worldUsers, isWorldUsersLoaded } = useWorldUsers();
 
-  const users = useSelector(usersSelector);
+  const { isCurrentVenueLoaded, currentVenue } = useConnectCurrentVenueNG(
+    venueId
+  );
 
-  return isLoaded(users);
+  return useMemo(
+    () => ({
+      recentVenueUsers:
+        isCurrentVenueLoaded && currentVenue
+          ? worldUsers.filter(
+              (user) => user.lastSeenIn?.[currentVenue.name] > lastSeenThreshold
```

Comment:
I have noticed that we have `lastSeenIn` set in milliseconds somewhere in the code.